pkg/models/gacha.go

```diff
@@ -22,42 +69,87 @@ func init() {
 	db.AutoMigrate(&User{})
 	db.Model(&User{}).ModifyColumn("name", "varchar(30)")
 	db.Model(&User{}).ModifyColumn("token", "char(30)")
+	db.AutoMigrate(&Character{})
+	db.Model(&Character{}).ModifyColumn("name", "varchar(30)")
+	db.AutoMigrate(&UserCharacter{})
+	db.AutoMigrate(&GachaResult{})
 }
 
 func (u *User) CreateUser() *User {
-	rand.Seed(time.Now().UnixNano())
-
 	u.Token = generateRandomString(30)
 	db.NewRecord(u)
 	db.Create(&u)
 	return u
 }
 
 // Function defined to generate random string for token
-
 func generateRandomString(length int) string {
 	const charset = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789"
+	charsetLength := big.NewInt(int64(len(charset)))
 	b := make([]byte, length)
-	for i := range b {
-		b[i] = charset[rand.Intn(len(charset))]
+	for i := 0; i < length; i++ {
+		randomIndex, _ := rand.Int(rand.Reader, charsetLength)
+		b[i] = charset[randomIndex.Int64()]
 	}
 	return string(b)
 }
 
-func GetAllUsers() []User {
-	var Users []User
-	db.Find(&Users)
-	return Users
+func GetUserByToken(token string) []User {
+	var users []User
+	db.Where("token = ?", token).Find(&users)
+	return users
+}
+
+func GetAllUsers(token string) []User {
```

Comment:
Is there situation where multiple users are returned with a single token?

Reply:
Yes initially I had just implemented with random characters  from user input in that case multiple users are returned for same token. However I later generated a token of 20 characters string which Is not  repeating in that case it gives unique output.

Reply:
@iamananya If it is clear that it is a unique Token, there is no need for a method to retrieve multiple users using a token.

Reply:
Okay.

Reply:
@GuiltyMorishita  I have commented that portion of code in  the latest commit for retrieving multiple users using a single token.